x/ibc-hooks/contracts/crosschain-swaps/README.md

```diff
@@ -0,0 +1,214 @@
+could enable support for any channel by specifying it as a parameter.
+
+The `slippage` can be set to a percentage of the twap price (as shown above), or as
+the minimum amount of tokens expected to be received: `{"min_output_amount": "100"}`.
+
+
+#### Optional keys
+
+If `track_ibc_sends` is enabled during instantiation, the `failed_delivery` key
+can be set to an address on Osmosis that will be allowed to recover the tokens
+in case of a failure. This key is optional and if ommited will default to
+`false`.
+
+The `next_memo` key, if provided, will be added to the IBC transfer as the memo
+for that transfer. This can be useful if the receiving chain also has IBC hooks
+on transfers. In that case, this can be used to specify how the receiver should
+deal with the received tokens (mostly useful when the receiver is a contract or
+another ibc actor).
+
+
+## Requirements
+
+To use this contract for crosschain swaps, the following are needed:
+
+ * The chain needs a wasm execute middleware that executes a contract when
+   receiving a wasm directive in the memo.
+ * The swaprouter contract should be instantiated
+ 
+Optional:
+
+This contract can be configured to track the acks or timeouts of the sent
+packets. For that we needed:
+
+ * The chain to provide a way to call a contract when the ack or timeout for a
+   packet is received. 
+ * Instantiate the contract with the `track_ibc_sends` set to true. Defaults
+   to false.
+
+
+## Testing
+
+The following is the procedure to test the contracts on a testnet.
+
+### Requirements
+
+You will need: 
+
+* An osmosis testnet
+* Another testnet with an open channel to the osmosis testnet
+* Tokens on the non-osmosis testnet
+* A pool on the osmosis testnet between the IBC'd native asset of the
+  non-osmosis testnet and osmo
+  
+### Example
+
+For the purpose of this test, we used the following testnets:
+
+Osmosis:
+
+``` toml
+# The network chain ID
+chain-id = "osmo-test-4"
+# The keyring's backend, where the keys are stored (os|file|kwallet|pass|test|memory)
+keyring-backend = "test"
+# CLI output format (text|json)
+output = "text"
+# <host>:<port> to Tendermint RPC interface for this chain
+node = "https://rpc-test.osmosis.zone:443"
+# Transaction broadcasting mode (sync|async|block)
+broadcast-mode = "block"
+```
+
+Gaia:
+
+``` toml
+# The network chain ID
+chain-id = "theta-testnet-001"
+# The keyring's backend, where the keys are stored (os|file|kwallet|pass|test|memory)
+keyring-backend = "test"
+# CLI output format (text|json)
+output = "json"
+# <host>:<port> to Tendermint RPC interface for this chain
+node = "http://seed-02.theta-testnet.polypore.xyz:26657"
+# Transaction broadcasting mode (sync|async|block)
+broadcast-mode = "block"
+```
+
+### Deploying the contracts
+
+Store the swaprouter code:
+
+``` sh
+> osmosisd-testnet tx wasm store ./bytecode/swaprouter.wasm --from owner  --gas auto --gas-prices 0.1uosmo --gas-adjustment 1.3 -y
```

Comment:
Note to self: I have better commands for this now. Cleanup and add the script I'm using to test this on the localrelayer tests.